From: Ajit Khaparde <ajit.khaparde@emulex.com>
Date: Thu, 26 Apr 2012 15:42:39 +0000
Subject: [PATCH 52/58] be2net: fix speed displayed by ethtool on certain SKUs

commit 2a89611a051cdcfcf4a7ae21d18eef64fab517b4 upstream.

logical speed returned by link_status_query needs to be multiplied by 10.

Signed-off-by: Ajit Khaparde <ajit.khaparde@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/emulex/benet/be_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/emulex/benet/be_ethtool.c b/drivers/net/ethernet/emulex/benet/be_ethtool.c
index 9d71bad..b3b5fae 100644
--- a/drivers/net/ethernet/emulex/benet/be_ethtool.c
+++ b/drivers/net/ethernet/emulex/benet/be_ethtool.c
@@ -557,7 +557,7 @@ static int be_get_settings(struct net_device *netdev, struct ethtool_cmd *ecmd)
 			if (!status)
 				be_link_status_update(adapter, link_status);
 			if (link_speed)
-				et_speed = link_speed;
+				et_speed = link_speed * 10;
 			else
 				et_speed = convert_to_et_speed(port_speed);
 		} else {
-- 
1.7.10

